execution/executor-types/src/lib.rs

```diff
@@ -151,9 +177,47 @@ impl StateComputeResult {
         &self.frozen_subtree_roots
     }
 
+    pub fn parent_num_leaves(&self) -> u64 {
+        self.parent_num_leaves
+    }
+
+    pub fn parent_frozen_subtree_roots(&self) -> &Vec<HashValue> {
+        &self.parent_frozen_subtree_roots
+    }
+
     pub fn has_reconfiguration(&self) -> bool {
         self.epoch_state.is_some()
     }
+
+    pub fn signature(&self) -> &Option<Ed25519Signature> {
+        &self.signature
+    }
+
+    pub fn set_signature(&mut self, sig: Ed25519Signature) {
+        self.signature = Some(sig);
+    }
+}
+
+/// A structure that wraps the state_compute_result of a block and the signature of the
+/// VoteProposal corresponding to the block.
+#[derive(Debug, PartialEq, Eq, Clone, Serialize, Deserialize)]
+pub struct StateComputeResultWithSignature {
```

Comment:
why do we have the signature here and in StateComputeResult?

Reply:
I think I forget to remove it.